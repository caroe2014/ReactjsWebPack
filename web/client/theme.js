// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
