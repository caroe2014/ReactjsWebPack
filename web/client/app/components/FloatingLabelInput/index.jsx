// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import { Input, Label, Flex, Text } from 'rebass';
import i18n from 'web/client/i18n';
import IconButton from 'web/client/app/components/IconButton';
import { AVOID_ONLY_FIRST_SPACE, FIRST_NAME_PATTERN } from 'web/client/app/utils/constants';

const InputWrapper = styled.div`
  position: relative;
  margin-bottom: 10px;
  min-height: 75px;
  width: auto;
`;

const InputParent = styled(Flex)`
  flex-direction: column-reverse;
  width: 100%;
  &>input::-ms-clear {
    display: none;
  }
  &>input::-webkit-inner-spin-button, 
  &>input::-webkit-outer-spin-button { 
    -webkit-appearance: none; 
    margin: 0; 
  }
`;

const InputText = styled(Input)`
  color: ${props => props.theme.colors.primaryText};
  font-size: 18px;
  left: 0;
  background-color: transparent;
  border: 0;
  border-radius: 0;
  border-bottom: 2px solid #EEE;
  padding: 3px 20px 3px 0px;
  box-shadow: none;
  outline: 0;
  padding-left: ${props => props.prefix ? '4px' : '0px'}
  &:disabled{
    cursor: not-allowed;
  }
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const PrefixCodeText = styled(Flex)`
  font-size: 18px;
  padding-bottom: 3px;
  font-weight: normal;
  color: ${props => props.theme.colors.primaryText};
  display: none;
  border-bottom: 2px solid #EEE;
  display:  ${props => props.length > 0 ? 'block' : 'none'}
  ${InputText}:focus ~ & {
    display: block};
  }
`;

const InputLabel = styled(Label)`
  pointer-events: none;
  color: ${props => props.theme.colors.primaryTextColor};
  ${props => {
    if (props.length > 0) {
      return `transform: translateY(8px) scale(0.85);
      font-weight: bold;
      color: ${props => props.theme.colors.primaryTextColor}};
      `;
    } else {
      return `transform: translateY(25px) scale(1);
      font-weight: normal;
      `;
    }
  }}
  transform-origin: left top;
  transition: 100ms;
  font-size: 16px;
  ${InputText}:focus ~ & {
    transform: translateY(8px) scale(0.85);
    color: ${props => props.theme.colors.primaryText};
    font-weight: bold;
  }
  ${props => {
    if (props.transformLabel) {
      return `transform: translateY(8px) scale(0.85);
      color: ${props => props.theme.colors.primaryText};
      font-weight: bold;`;
    }
  }}

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const InputPlaceholder = styled(Text)`
  pointer-events: none;
  position: absolute;
  color: ${props => props.theme.colors.textGrey};
  opacity: 0.75;
  transform-origin: left top;
  transition: 100ms;
  font-size: 16px;
  display: none;
  bottom: 28px;
  left: 0px;
  left: ${props => props.prefix ? ((props.prefix.length) * 12) + 'px' : '0px'};
  margin-bottom: 0px;
  ${InputText}:focus ~ & {
    display: ${props => props.length > 0 ? 'none' : 'block'};
  }
  ${props => {
    if (props.transformLabel && props.length > 0) {
      return `display: none`;
    } else {
      return props.transformLabel ? 'display: block;' : 'display: none;';
    }
  }}
`;

const CustomInputParent = styled(Flex)`
  flex-direction: row-reverse;
  align-items: flex-end;
`;

const ErrorMessage = styled.p`
  color: ${props => props.theme.colors.validationError};
  height: auto;
  font-size: 14px;
  min-height: 1.2em;
  padding-left: 3px;
  margin: 5px 0px 0px 0px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const CloseButton = styled(props => <IconButton {...props} />)`
  &:focus {
    outline: none;
    box-shadow: none;
  }
  display: ${props => props['disabled'] ? 'none' : 'block'};
    bottom: 25px;
    font-size: ${props => props.theme.fontSize.md};
    right:0;
    color: ${props => props.theme.colors.textGrey};
    font-weight: 500;
    position: absolute;
    opacity: ${props => {
    if (props.disabled) return '0.6 !important';
    return 'pointer';
  }};
  cursor: ${props => {
    if (props.disabled) return 'not-allowed';
    return 'pointer';
  }};
`;

class FloatingLabelInput extends Component {

  constructor (props) {
    super(props);
    this.state = {
      transformLabel: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.onClearInput = this.onClearInput.bind(this);
    this.onFocus = this.onFocus.bind(this);
    this.onBlur = this.onBlur.bind(this);
  }

  onFocus () {
    this.setState({ transformLabel: true });
  }

  onBlur () {
    this.setState({ transformLabel: false });
  }

  handleChange = event => {
    const { callBack, propertyName, validationRegEx } = this.props;
    const textValue = event.target.value;
    if (validationRegEx && textValue.length > 0) {
      if ((validationRegEx === AVOID_ONLY_FIRST_SPACE || validationRegEx === FIRST_NAME_PATTERN) && textValue.length > 30) { // eslint-disable-line max-len
        return;
      }
      const regExp = new RegExp(validationRegEx);
      if (!regExp.test(textValue)) {
        return;
      }
    }
    callBack && callBack(propertyName, textValue);
  }

  onClearInput () {
    const { callBack, propertyName } = this.props;
    callBack && callBack(propertyName, '');
  }

  render () {
    const { label, ariaLabel, ariaInvalid, tabIndex, placeHolder, value, error, inputType,
      clearIcon, pattern, disabled, className, prefix } = this.props;
    return (
      <ThemeProvider theme={theme}>
        <InputWrapper className='input-field-wrapper'>
          <InputParent className='input-field-parent'>
            <CustomInputParent className='text-div' id='parent'>
              <InputText className={className ? `input-text ${className}` : 'input-text'} onChange={this.handleChange}
                value={value} type={inputType || 'text'}
                aria-label={ariaLabel}
                aria-invalid={ariaInvalid}
                tabIndex={tabIndex}
                pattern={pattern || ''}
                disabled={disabled}
                prefix={prefix}
                prefixWidth={this.prefix && this.prefix.offsetWidth}
                onFocus={this.onFocus}
                onBlur={this.onBlur}
              />
              {prefix && <PrefixCodeText length={value.length}>{prefix}</PrefixCodeText>}
            </CustomInputParent>
            <InputLabel
              className='input-field-label'
              length={value.length}
              prefix={prefix}
              transformLabel={this.state.transformLabel}>
              {label}
            </InputLabel>
            {placeHolder &&
              <InputPlaceholder
                className='input-placeholder'
                prefix={prefix}
                transformLabel={this.state.transformLabel}
                length={value.length}>
                {placeHolder}
              </InputPlaceholder>
            }
            {clearIcon &&
              <CloseButton children='&#10005;'
                tabIndex={0} aria-label={i18n.t('CLEAR_VALUE')} role='button'
                onClick={!disabled ? this.onClearInput : undefined}
                className='clear-value'
                disabled={value && value.length > 0 ? 0 : -1}
              />}
          </InputParent>
          <ErrorMessage tabIndex={error && error ? 0 : -1}
            aria-live='polite' className='input-field-error'>{error && error}</ErrorMessage>
        </InputWrapper>
      </ThemeProvider>
    );
  }
}

export default FloatingLabelInput;
