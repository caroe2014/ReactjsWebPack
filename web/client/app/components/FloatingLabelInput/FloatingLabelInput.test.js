import React from 'react';
import {
  mount,
  shallow,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';

import FloatingLabelInput from '.';
configure({
  adapter: new Adapter()
});

describe('FloatingLabelInput', () => {

  let callback = jest.fn();

  it('does a snapshot test', () => {
    const wrapper = mount(<ThemeProvider theme={theme}><FloatingLabelInput value='test'/></ThemeProvider>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(mount(<ThemeProvider theme={theme}><FloatingLabelInput placeHolder='test' value='test'/></ThemeProvider>).exists()).toBe(true);
  });

  it('should render InputLabel with normal font', () => {
    expect(mount(<ThemeProvider theme={theme}><FloatingLabelInput value='' label={'testLabel'} clearIcon callBack={callback} propertyName={'testProperty'}/></ThemeProvider>).exists()).toBe(true);
  });

  it('should render InputLabel with normal font', () => {
    expect(mount(<ThemeProvider theme={theme}><FloatingLabelInput value='' placeHolder='test' label={'testLabel'} clearIcon callBack={callback} propertyName={'testProperty'}/></ThemeProvider>).exists()).toBe(true);
  });

  it('should call  onClearInput with mock', () => {
    let wrapper = shallow(<FloatingLabelInput value='test' label={'testLabel'} clearIcon callBack={callback} propertyName={'testProperty'}/>);
    let mockFn = jest.fn();
    wrapper.instance().onClearInput = mockFn;
    wrapper.instance().forceUpdate();
    wrapper.update();
    wrapper.find('.clear-value').simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should call onClearInput with spy', () => {
    const callback = jest.fn();
    let wrapper = shallow(<FloatingLabelInput value='' placeHolder='test' label={'testLabel'} clearIcon callBack={callback} propertyName={'testProperty'}/>);
    const spy = jest.spyOn(wrapper.instance(), 'onClearInput');
    wrapper.instance().forceUpdate();
    wrapper.find('.clear-value').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call handleChange when InputText is entered', () => {
    const callback = jest.fn();
    let wrapper = shallow(<FloatingLabelInput value='' placeHolder='test' label={'testLabel'} clearIcon callBack={callback} propertyName={'testProperty'}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    wrapper.find('.input-text').simulate('focus');
    wrapper.find('.input-text').simulate('change', { target: { value: 'tests' } });
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call handleChange when InputText is entered when regex is present', () => {
    const callback = jest.fn();
    let wrapper = shallow(<FloatingLabelInput value='' placeHolder='test' label={'testLabel'} validationRegEx={'someRegex'} clearIcon callBack={callback} propertyName={'testProperty'}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    wrapper.find('.input-text').simulate('focus');
    wrapper.find('.input-text').simulate('change', { target: { value: 'tests' } });
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

});
