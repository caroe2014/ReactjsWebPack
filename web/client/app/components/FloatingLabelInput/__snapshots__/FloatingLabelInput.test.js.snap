// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingLabelInput does a snapshot test 1`] = `
.c3 {
  box-sizing: border-box;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 14px;
  margin-bottom: 4px;
}

.c7 {
  font-size: inherit;
  line-height: inherit;
  width: 100%;
  border: 0;
  border-color: gray;
  box-shadow: 1;
  border-radius: 2px;
  margin: 0px;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: inherit;
  background-color: transparent;
  font-family: inherit;
  display: inline-block;
  vertical-align: middle;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c7:focus {
  outline: none;
  box-shadow: inset 0 0 0 1px undefined;
}

.c7:disabled {
  opacity: 0.25;
}

.c0 {
  position: relative;
  margin-bottom: 10px;
  min-height: 75px;
  width: auto;
}

.c1 {
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  width: 100%;
}

.c1>input::-ms-clear {
  display: none;
}

.c1>input::-webkit-inner-spin-button,
.c1>input::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c6 {
  color: #383838;
  font-size: 18px;
  left: 0;
  background-color: transparent;
  border: 0;
  border-radius: 0;
  border-bottom: 2px solid #EEE;
  padding: 3px 20px 3px 0px;
  box-shadow: none;
  outline: 0;
  padding-left: 0px;
}

.c6:disabled {
  cursor: not-allowed;
}

.c6:focus {
  outline: none;
  box-shadow: none;
}

.c8 {
  pointer-events: none;
  -webkit-transform: translateY(8px) scale(0.85);
  -ms-transform: translateY(8px) scale(0.85);
  transform: translateY(8px) scale(0.85);
  font-weight: bold;
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transition: 100ms;
  transition: 100ms;
  font-size: 16px;
}

.c8 color:function (props) {
  return: props.theme.colors.primaryTextColor;
}

.c5:focus ~ .c8 {
  -webkit-transform: translateY(8px) scale(0.85);
  -ms-transform: translateY(8px) scale(0.85);
  transform: translateY(8px) scale(0.85);
  color: #383838;
  font-weight: bold;
}

.c8:focus {
  outline: none;
  box-shadow: none;
}

.c4 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c10 {
  color: #cc0000;
  height: auto;
  font-size: 14px;
  min-height: 1.2em;
  padding-left: 3px;
  margin: 5px 0px 0px 0px;
}

.c10:focus {
  outline: none;
  box-shadow: none;
}

<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <FloatingLabelInput
    value="test"
  >
    <ThemeProvider
      theme={
        Object {
          "colors": Object {
            "addOnTab": "#347CFF",
            "addToCart": "#347CFF",
            "addonBg": "#EEEEEE",
            "backgroundGrey": "#E6E6E6",
            "black": "#000000",
            "close": "red",
            "contextBackground": "#f0f0f3",
            "darkGrey": "#464646",
            "disableBackground": "#E4E4E4",
            "disableButton": "#D1D1D1",
            "error": "#cc0000",
            "footer": "#48a942",
            "glutenFree": "#8d7356",
            "healthy": "#359919",
            "hint": "#a6a6a6",
            "inputBottomBorder": "lightgray",
            "itemBorder": "lightgrey",
            "itemTile": "#3C474C0",
            "light": "white",
            "loader": "#ababab",
            "loginBackground": "#303030",
            "loginBoxColor": "#434149",
            "loginField": "#AAA7AC",
            "loginFormBackground": "#484848",
            "navigationMenu": "#48a942",
            "navigationMenuHover": "cornflowerblue",
            "new": "#ff6000",
            "open": "#48a942",
            "paymentSuccessTitle": "#505050",
            "placeHolder": "#909090",
            "primary": "#347CFF",
            "primaryButton": "#347CFF",
            "primaryText": "#383838",
            "secondaryText": "#a6a6a6",
            "spicy": "#ed1c24",
            "text": "black",
            "textGrey": "#505050",
            "tile": "#505050",
            "validationError": "#cc0000",
            "vegan": "#0d4a22",
            "vegetarian": "#0d4a22",
          },
          "fontFamily": "Roboto",
          "fontSize": Object {
            "lg": "1.8rem",
            "md": "1.3rem",
            "nm": "1rem",
            "sm": "0.75rem",
            "xl": "2.4rem",
          },
          "fontSizeFromProps": [Function],
          "mediaBreakpoints": Object {
            "desktop": [Function],
            "mobile": [Function],
            "tablet": [Function],
          },
          "nav": Object {
            "height": "70px",
          },
          "receiptFontFamily": "Roboto",
          "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
          "spacing": Object {
            "lg": "4em",
            "md": "2em",
            "nm": "1em",
            "none": "0em",
            "sm": "0.5em",
            "xl": "8em",
          },
        }
      }
    >
      <styled.div
        className="input-field-wrapper"
      >
        <div
          className="input-field-wrapper c0"
        >
          <Styled(Flex)
            className="input-field-parent"
          >
            <Flex
              className="input-field-parent c1"
            >
              <Box
                className="input-field-parent c1 c2"
              >
                <div
                  className="input-field-parent c1 c2 c3"
                >
                  <Styled(Flex)
                    className="text-div"
                    id="parent"
                  >
                    <Flex
                      className="text-div c4"
                      id="parent"
                    >
                      <Box
                        className="text-div c4 c2"
                        id="parent"
                      >
                        <div
                          className="text-div c4 c2 c3"
                          id="parent"
                        >
                          <Styled(Input)
                            className="input-text"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            pattern=""
                            type="text"
                            value="test"
                          >
                            <Input
                              bg="transparent"
                              border={0}
                              borderColor="gray"
                              borderRadius={2}
                              boxShadow={1}
                              className="input-text c5 c6"
                              color="inherit"
                              fontSize="inherit"
                              is="input"
                              lineHeight="inherit"
                              m={0}
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              pattern=""
                              px={1}
                              py={2}
                              type="text"
                              value="test"
                              width={1}
                            >
                              <ForwardRef
                                bg="transparent"
                                border={0}
                                borderColor="gray"
                                borderRadius={2}
                                boxShadow={1}
                                className="input-text c5 c6 c7"
                                color="inherit"
                                fontSize="inherit"
                                is="input"
                                lineHeight="inherit"
                                m={0}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                pattern=""
                                px={1}
                                py={2}
                                type="text"
                                value="test"
                                width={1}
                              >
                                <input
                                  className="input-text c5 c6 c7"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  pattern=""
                                  type="text"
                                  value="test"
                                />
                              </ForwardRef>
                            </Input>
                          </Styled(Input)>
                        </div>
                      </Box>
                    </Flex>
                  </Styled(Flex)>
                  <Styled(Label)
                    className="input-field-label"
                    length={4}
                    transformLabel={false}
                  >
                    <Label
                      alignItems="center"
                      className="input-field-label c8"
                      fontSize={1}
                      is="label"
                      length={4}
                      mb={1}
                      transformLabel={false}
                    >
                      <ForwardRef
                        alignItems="center"
                        className="input-field-label c8 c9"
                        fontSize={1}
                        is="label"
                        length={4}
                        mb={1}
                        transformLabel={false}
                      >
                        <label
                          className="input-field-label c8 c9"
                          length={4}
                          transformLabel={false}
                        />
                      </ForwardRef>
                    </Label>
                  </Styled(Label)>
                </div>
              </Box>
            </Flex>
          </Styled(Flex)>
          <styled.p
            aria-live="polite"
            className="input-field-error"
            tabIndex={-1}
          >
            <p
              aria-live="polite"
              className="input-field-error c10"
              tabIndex={-1}
            />
          </styled.p>
        </div>
      </styled.div>
    </ThemeProvider>
  </FloatingLabelInput>
</ThemeProvider>
`;
